File: hash.java
import java.util.HashMap;

public class SimpleExample {
    public static void main(String[] args) {
        // Используем готовую реализацию HashMap
        HashMap<String, Integer> map = new HashMap<>();
        
        // Добавляем элементы
        map.put("apple", 5);
        map.put("banana", 3);
        map.put("orange", 8);
        
        // Выводим содержимое
        System.out.println("HashMap: " + map);
        
        // Получаем значения
        System.out.println("apple -> " + map.get("apple"));
        System.out.println("banana -> " + map.get("banana"));
        
        // Обновляем значение
        map.put("apple", 10);
        System.out.println("updated apple -> " + map.get("apple"));
        
        // Проверяем наличие ключа
        System.out.println("Contains 'orange': " + map.containsKey("orange"));
    }
}

File: hash.cpp
#include <iostream>
#include <unordered_map>

// Используем готовую реализацию из STL
int main() {
    std::unordered_map<std::string, int> ht;

    // Добавляем элементы
    ht["apple"] = 5;
    ht["banana"] = 3;
    ht["orange"] = 8;

    // Выводим содержимое
    for (const auto& pair : ht) {
        std::cout << pair.first << " -> " << pair.second << std::endl;
    }

    // Получаем значения
    std::cout << "apple -> " << ht["apple"] << std::endl;
    std::cout << "banana -> " << ht.at("banana") << std::endl;

    return 0;
}
